release-application.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+# This script should behave like maven release plugin - merge the current working into master, make sure the version does not contain
+# any SNAPSHOT dependencies, perform the release, prepares the next iteration. It supports working in hotfix mode - where
+# a patched version is released.
+
+# Exit on any error to avoid troubles.
+# Print the statements before running them
+set -ex
+
+RELEASE_PROCESS_PROPERTIES="release-application.properties"
+APPLICATION_PROPERTIES="src/main/resources/application.properties"
+APPLICATION_VERSION_PROPERTY="application_version"
+
+extract_property() {
+    local PROPERTIES_FILE=$1
+    local PROPERTY_NAME=$2
+    PROPERTY_VALUE=`grep "^${PROPERTY_NAME}\s*=" "${PROPERTIES_FILE}" | sed "s/${PROPERTY_NAME}\s*=\s*//"`
+
+    if [ -z  ${PROPERTY_VALUE}  ]
+    then
+        echo "Could not found property '${PROPERTY_NAME}' in file '${PROPERTIES_FILE}'!"
+        exit 1
+    fi
+}
+
+extract_release_branch_var() {
+    extract_property ${RELEASE_PROCESS_PROPERTIES} "releaseBranch"
+    RELEASE_BRANCH_NAME=${PROPERTY_VALUE}
+}
+
+extract_is_hotfix_var() {
+    extract_property ${RELEASE_PROCESS_PROPERTIES} "hotfix"
+    if [ ${PROPERTY_VALUE} = "yes" ]
+    then
+        HOTFIX=1
+    else
+        HOTFIX=0
+    fi
+}
+
+extract_development_branch_var() {
+    extract_property ${RELEASE_PROCESS_PROPERTIES} "developmentBranch"
+    DEVELOPMENT_BRANCH_NAME=${PROPERTY_VALUE}
+
+    if [ ${HOTFIX} = 1 ]
+    then
+        extract_property ${RELEASE_PROCESS_PROPERTIES} "hotfixBranch"
+        DEVELOPMENT_BRANCH_NAME=${PROPERTY_VALUE}
+    fi
+}
+
+extract_build_profile_var() {
+    extract_property ${RELEASE_PROCESS_PROPERTIES} "buildProfile"
+    BUILD_PROFILE=${PROPERTY_VALUE}
+}
+
+extract_application_versions() {
+
+    # Extract the current application version
+    local CURRENT_VERSION=`printf 'APPLICATION_VERSION=${project.version}\n0\n' |\
+        mvn help:evaluate |\
+        grep '^APPLICATION_VERSION=' |\
+        sed -r 's/(^APPLICATION_VERSION=)(.+)(-SNAPSHOT|-RELEASE)/\2/'`
+
+    RELEASED_VERSION=${CURRENT_VERSION}-RELEASE
+
+    # Extract the versions
+    local MAJOR=`echo ${CURRENT_VERSION} | sed -r 's/([0-9]+).([0-9]+).([0-9]+)/\1/'`
+    local MINOR=`echo ${CURRENT_VERSION} | sed -r 's/([0-9]+).([0-9]+).([0-9]+)/\2/'`
+    local PATCH=`echo ${CURRENT_VERSION} | sed -r 's/([0-9]+).([0-9]+).([0-9]+)/\3/'`
+
+    if [ ${HOTFIX} = 1 ]
+    then
+        RELEASED_VERSION="${MAJOR}.${MINOR}.$((PATCH+1))-RELEASE"
+    else
+        NEXT_VERSION="${MAJOR}.$((MINOR+1)).0-SNAPSHOT"
```

Comment:
how are we going to increment the major version if we always increment minor? 



> Major version X (X.y.z | X > 0) MUST be incremented if any backwards incompatible changes are introduced to the public API. It MAY include minor and patch level changes. Patch and minor version MUST be reset to 0 when major version is incremented.

https://semver.org/